### PR TITLE
86956 claims information section - form 10-7959a

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR adds the claims information section to form 10-7959a. It includes several new screens, most of which are file uploads.

- Refactors some shared code to make it more flexible across forms
- Adds claims screener questions (medical/pharmacy + work/auto)
- Adds file upload screens to claims info section
- Team: IVC CHAMPVA
- Flipper: NA

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86956

## Testing done

- Manual
- Verified unit tests run and pass

## Screenshots

|         |  Screens |
| ------- |  ----- |
| Medical/Pharmacy |Screenshot 2024-07-08 at 09 53 12|
| Work related |Screenshot 2024-07-08 at 09 53 08|
| Auto related | Screenshot 2024-07-08 at 09 53 04 |
| Medical supporting docs |Screenshot 2024-07-08 at 09 52 39|
| Explanation of Benefits |Screenshot 2024-07-08 at 09 52 47|
| Additional info |Screenshot 2024-07-08 at 09 52 56|
| Pharmacy claim upload |Screenshot 2024-07-08 at 09 52 17|
|Review page|Screenshot 2024-07-08 at 09 53 40|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

File: src/applications/ivc-champva/10-7959a/chapters/claimInformation.js [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+export const eobUploadSchema = isPrimary => {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/chapters/claimInformation.js:186]: This page schema is used for the two EOB upload screens - by passing `true` or `false` we control which health insurance company name is shown in the header.

File: src/applications/ivc-champva/10-7959a/chapters/claimInformation.js
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/chapters/claimInformation.js:1]: All new page schemas are exported from here

File: src/applications/ivc-champva/shared/components/fileUploads/attachments.js [hunk 2]
```diff
@@ -34,26 +34,35 @@ export const fileWithMetadataSchema = (possibleFiles, minItems = 1) => {
   };
 };
 
+export const fileUploadBlurbCustom = (items, afterContent) => {
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/attachments.js:37]: Reworked this generic piece of markup to accept some customization arguments.

File: src/applications/ivc-champva/shared/components/fileUploads/upload.js [hunk 3]
```diff
@@ -67,7 +67,12 @@ export const fileUploadUi = content => {
       },
     }),
     // hideOnReview: true,
-    // attachmentName: true,
+    // attachmentName: false,
+    attachmentName: content?.attachmentName
+      ? {
+          'ui:title': 'Document name',
+        }
+      : false,
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/upload.js:75]: lets us pass the `attachmentName` prop to toggle showing the editable filename text field while maintaining backward compatibility with other forms' usage of this upload component.

**Review by balexandr — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Review by IGallupSoCo — APPROVED** [review]
Loving the clear code comments!  LGTM

**Pull request merged**